platform-sdk/swirlds-common/build.gradle.kts

```diff
@@ -15,43 +15,27 @@
  */
 
 plugins {
-    id("com.swirlds.platform.conventions")
-    `java-library`
-    id("com.swirlds.platform.maven-publish")
-    id("org.gradle.java-test-fixtures")
+    id("com.hedera.hashgraph.sdk.conventions")
+    id("com.hedera.hashgraph.maven-publish")
+    id("java-test-fixtures")
 }
 
 dependencies {
-    // Individual Dependencies
-    api(project(":swirlds-base"))
-    api(project(":swirlds-logging"))
-    api(project(":swirlds-config-api"))
-    implementation(libs.classgraph)
-    implementation(libs.commons.codec)
-    implementation(libs.prometheus.httpserver) {
-        exclude("io.prometheus", "simpleclient_tracer_otel")
-        exclude("io.prometheus", "simpleclient_tracer_otel_agent")
-    }
-    compileOnly(libs.spotbugs.annotations)
-
-    // Bundle Dependencies
-    api(libs.bundles.cryptography.core)
-    runtimeOnly(libs.bundles.cryptography.runtime)
-    implementation(libs.bundles.logging.impl)
-    compileOnly(libs.spotbugs.annotations)
+    javaModuleDependencies {
+        runtimeOnly(gav("resource.loader"))
+        runtimeOnly(gav("com.sun.jna"))
 
-    // Test Dependencies
-    testCompileOnly(libs.spotbugs.annotations)
-    testImplementation(testLibs.bundles.junit)
-    testImplementation(testLibs.bundles.mocking)
-    testImplementation(testLibs.bundles.utils)
-    testImplementation(testFixtures(project(":swirlds-base")))
-    testImplementation(project(":swirlds-config-impl"))
-    testImplementation(project(":swirlds-unit-tests:common:swirlds-test-framework"))
-    testImplementation(project(":swirlds-unit-tests:common:swirlds-common-test"))
-    testImplementation(testFixtures(project(":swirlds-config-api")))
-
-    // Test Fixtures
-    testFixturesImplementation(testLibs.bundles.junit)
-    testFixturesCompileOnly(libs.spotbugs.annotations)
+        testImplementation(project(":swirlds-common"))
```

Comment:
Good catch. I agree that this does not look right. If you consider the tests an extension of the "main" sources (whitebox testing) there should be no need to declare this.

Here, I followed strictly what the _dependency analysis plugin_ suggested. I think the reason why this is here, is because there are "test-fixtures". And typically the dependency order is.

"main" <- "test fixtures" <- "test" 

In the module setup, we kind of merge "main" and "test" and then conceptually have this:

"test fixtures" <- "patchedModule(main, test)"

I think I could just remove this dependency here and it will still work. Just the analysis plugin will complain. I'll check.